### PR TITLE
add Cow<str> conversion to Value

### PR DESCRIPTION
**Pull request by beckend**
## PR Info

- Closes 
https://github.com/SeaQL/sea-query/issues/548

- Dependencies:
  - use std::borrow::Cow;

## New Features

- [x] 
Value can now be converted from `Cow<str>`

### CODE REVIEW

File: src/value.rs [hunk 1]
```diff
@@ -435,6 +437,37 @@ where
...
+where
+    T: Clone + AsRef<str> + Into<Value>,
+    Cow<'a, T>: From<Cow<'a, str>>,
+{
+    fn from(x: Cow<'a, T>) -> Value {
+        x.as_ref().clone().into()
+    }
+}
+
+impl<'a> ValueType for Cow<'a, str> {
```
Comment (ikrivosheev) [inline src/value.rs:448]: Why you need this implementation?

File: src/value.rs [hunk 2]
```diff
@@ -435,6 +437,35 @@ where
     }
 }
 
+impl<'a> From<Cow<'a, str>> for Value {
+    fn from(x: Cow<'a, str>) -> Value {
+        // will stack overflow if not cloning the ref
+        #[allow(clippy::clone_double_ref)]
+        x.as_ref().clone().into()
```
Comment (ikrivosheev) [inline src/value.rs:444]: `x.into_owned().into()`? And you don`t need to skip clippy warning

**Review by ikrivosheev — COMMENTED** [review]
@beckend, hey! Thank you for the PR! Some comments.

File: src/value.rs (hunk 1 above)
Comment (beckend) [reply src/value.rs:448]:
```shell
    --> src/value.rs:1488:24
     |
1488 |         let v: Value = val.into();
     |                        ^^^ ---- required by a bound introduced by this call
     |                        |
     |                        the trait `From<std::borrow::Cow<'_, str>>` is not implemented for `value::Value`
```

File: src/value.rs (hunk 2 above)
Comment (beckend) [reply src/value.rs:444]: I simply did not know about it, will fix

File: src/value.rs (hunk 1 above)
Comment (beckend) [reply src/value.rs:446]: sorry, that is the other one, this one is ValueType
```shell
--> src/value.rs:1474:31
     |
1474 |         let out: Cow<str> = v.unwrap();
     |                               ^^^^^^ the trait `value::ValueType` is not implemented for `std::borrow::Cow<'_, str>`
```

**Review by ikrivosheev — APPROVED** [review]
@beckend thank you! LGTM! @billy1624 @tyt2y3 can someone review PR?

File: src/value.rs [hunk 3]
```diff
@@ -435,6 +437,33 @@ where
...
+impl<'a> From<Cow<'a, str>> for Value {
+    fn from(x: Cow<'a, str>) -> Value {
+        x.into_owned().into()
+    }
+}
+
+impl<'a> ValueType for Cow<'a, str> {
+    fn try_from(v: Value) -> Result<Self, ValueTypeErr> {
+        match v {
+            Value::String(Some(x)) => Ok(x.into()),
```
[the remark this answers is not in the document]
Comment (beckend) [reply src/value.rs:449]: you mean x?

[the remark this answers is not in the document]
Comment (ikrivosheev) [reply src/value.rs:449]: Yes

File: src/value.rs [hunk 4]
```diff
@@ -435,6 +437,33 @@ where
...
+impl<'a> From<Cow<'a, str>> for Value {
+    fn from(x: Cow<'a, str>) -> Value {
+        x.into_owned().into()
+    }
+}
+
+impl<'a> ValueType for Cow<'a, str> {
+    fn try_from(v: Value) -> Result<Self, ValueTypeErr> {
+        match v {
+            Value::String(Some(x)) => Ok(*x.into()),
```
Comment (ikrivosheev) [inline src/value.rs:449]:
```suggestion
            Value::String(Some(x)) => Ok((*x).into()),
```

**Review by ikrivosheev — APPROVED** [review]

**Review by billy1624 — COMMENTED** [review]
Hey @beckend, thanks for contributions!

Just a small nitpick: https://github.com/beckend/sea-query/pull/1

**Review by billy1624 — APPROVED** [review]
Thanks!! @beckend

**Pull request merged**